### DISCUSSION TITLE
Sponsoring fixed amount, spread to all an orgs dependencies

### DISCUSSION THREAD

**Discussion by Danack** [community/community#8787] [Other Feature Feedback, Questions, & Ideas]
Hi,

Over the past 6 months or so, I have been encouraging/berating companies into sponsoring open source projects.

One common piece of feedback I get is that it's still too hard* for small companies to make appropriate sized sponsorships. e.g. the accounts department doesn't know what dependencies the projects in their company use, and even if they did, going through all of them and setting up individual sponsorships would be time consuming.

I think it would be great if:

* companies could set their sponsoring rate at say $50 per month.
* some tool either automatically, or guided by one of their devs finds all of their dependencies that are currently in use, and retrieves the sponsoring locations. e.g. for PHP the command `php composer.phar fund -f json` lists all of the funding possibilities and I believe `npm fund` also exists.
* the sponsoring money is split between all their dependencies.

Separately, but related, it would be nice if there was a way for developers to see if companies were supporting open source development, or whether they are leachers. i.e. is that company sponsoring an amount that is kind of inline with their revenue, or are they giving a lot less than average.

cheers
Dan
Ack

* I am slightly cynical about it ever being easy enough, but I can see that it could be made easier right now.

**Comment by jhassine** [comment]
Github already has the dependency graph feature that would nicely support the implementation of this idea as to identify the parties to pay.

Github also has user's payment card details for billing.

The only missing piece of the idea would be how to distribute the received payments to the developers?

**Reply by Danack** [reply]
> The only missing piece of the idea would be how to distribute the received payments to the developers?

Probably evenly. (assuming you meant distribute in the sense of divide, rather than in the sense of send.)

I personally would much prefer to see some maintainers receive more than a perfectly 'fair' share of money, than the current situation of every maintainer getting paid thousands of orders magnitude less than the value they provide to the world.

As a point of reference, I would probably be one of the maintainers who would 'miss out on a fair share', as most PHP projects don't bother to list which PHP extensions they use, instead only listing userland libraries written in PHP, as the userland libraries are the ones that are installed by Composer, and PHP extensions are not.

/still getting used to these new discussions...

**Comment by Draken1943** [comment]
> Hi,
> 
> Over the past 6 months or so, I have been encouraging/berating companies into sponsoring open source projects.
> 
> One common piece of feedback I get is that it's still too hard* for small companies to make appropriate sized sponsorships. e.g. the accounts department doesn't know what dependencies the projects in their company use, and even if they did, going through all of them and setting up individual sponsorships would be time consuming.
> 
> I think it would be great if:
> 
> * companies could set their sponsoring rate at say $50 per month.
> * some tool either automatically, or guided by one of their devs finds all of their dependencies that are currently in use, and retrieves the sponsoring locations. e.g. for PHP the command `php composer.phar fund -f json` lists all of the funding possibilities and I believe `npm fund` also exists.
> * the sponsoring money is split between all their dependencies.
> 
> Separately, but related, it would be nice if there was a way for developers to see if companies were supporting open source development, or whether they are leachers. i.e. is that company sponsoring an amount that is kind of inline with their revenue, or are they giving a lot less than average.
> 
> cheers Dan Ack
> 
> * I am slightly cynical about it ever being easy enough, but I can see that it could be made easier right now.

**Reply by Draken1943** [reply]
I think it would be better if a company that wants to work together understands and is aware that the services we provide have standards, and before starting to work together it has been agreed
